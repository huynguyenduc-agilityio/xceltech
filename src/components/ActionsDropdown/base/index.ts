export { DropdownMenuRoot } from './DropdownMenuRoot';
export { DropdownMenuTrigger } from './DropdownMenuTrigger';
export { DropdownMenuContent } from './DropdownMenuContent';
export { DropdownMenuItem } from './DropdownMenuItem';
