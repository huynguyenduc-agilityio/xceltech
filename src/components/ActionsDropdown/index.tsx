import { cn } from '@/utils';

import {
  DropdownMenuRoot,
  DropdownMenuTrigger,
  DropdownMenuContent,
  DropdownMenuItem,
} from './base';

export interface ActionsDropdownProps {
  children: React.ReactNode;
  items: { label: string; onClick?: () => void; disabled?: boolean }[];
  className?: string;
  itemClassName?: string;
}

const ActionsDropdown = ({
  children,
  items,
  className,
  itemClassName,
}: ActionsDropdownProps) => {
  return (
    <DropdownMenuRoot>
      <DropdownMenuTrigger asChild>{children}</DropdownMenuTrigger>
      <DropdownMenuContent className={cn(className)}>
        {items.map((item) => (
          <DropdownMenuItem
            key={item.label}
            onClick={item.onClick}
            disabled={item.disabled}
            className={cn(itemClassName)}
          >
            {item.label}
          </DropdownMenuItem>
        ))}
      </DropdownMenuContent>
    </DropdownMenuRoot>
  );
};

export default ActionsDropdown;
