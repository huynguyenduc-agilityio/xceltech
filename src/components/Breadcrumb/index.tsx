import React from 'react';
import { Link } from 'react-router-dom';

type BreadcrumbItem = {
  label: string;
  href?: string;
};

type BreadcrumbProps = {
  items: BreadcrumbItem[];
};

const Breadcrumb = ({ items }: BreadcrumbProps) => {
  return (
    <nav className="flex items-center gap-2 text-xl text-black-soft">
      {items.map((item, index) => (
        <React.Fragment key={index}>
          {item.href ? (
            <Link to={item.href} className="hover:underline">
              {item.label}
            </Link>
          ) : (
            <span>{item.label}</span>
          )}
          {index < items.length - 1 && '>'}
        </React.Fragment>
      ))}
    </nav>
  );
};

export default Breadcrumb;
