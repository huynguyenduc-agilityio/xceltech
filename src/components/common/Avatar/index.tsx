import * as React from 'react';
import * as AvatarPrimitive from '@radix-ui/react-avatar';

import { cn, generateBgColor } from '@/utils';

const AvatarRoot = React.forwardRef<
  React.ElementRef<typeof AvatarPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Root>
>(({ className, ...props }, ref) => (
  <AvatarPrimitive.Root
    ref={ref}
    className={cn(
      'relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full',
      className,
    )}
    {...props}
  />
));
AvatarRoot.displayName = AvatarPrimitive.Root.displayName;

const AvatarImage = React.forwardRef<
  React.ElementRef<typeof AvatarPrimitive.Image>,
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Image>
>(({ className, ...props }, ref) => (
  <AvatarPrimitive.Image
    ref={ref}
    className={cn('aspect-square h-full w-full', className)}
    {...props}
  />
));
AvatarImage.displayName = AvatarPrimitive.Image.displayName;

const AvatarFallback = React.forwardRef<
  React.ElementRef<typeof AvatarPrimitive.Fallback>,
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Fallback>
>(({ className, ...props }, ref) => (
  <AvatarPrimitive.Fallback
    ref={ref}
    className={cn(
      'flex h-full w-full items-center justify-center rounded-full bg-muted text-2xl',
      className,
    )}
    {...props}
  />
));
AvatarFallback.displayName = AvatarPrimitive.Fallback.displayName;

type AvatarContainerProps = {
  src: string;
  alt?: string;
  fallback?: React.ReactNode;
  size?: number | string;
  className?: string;
  fallbackClass?: string;
  onClick?: () => void;
};

const Avatar = ({
  src,
  fallback,
  alt = 'avatar',
  size = 40,
  className,
  fallbackClass,
  onClick,
}: AvatarContainerProps) => {
  const computedSize = typeof size === 'number' ? `${size}px` : size;
  const bgColor = generateBgColor(alt);

  return (
    <AvatarRoot
      style={{ width: computedSize, height: computedSize }}
      className={className}
      onClick={onClick}
    >
      <AvatarImage alt={alt} src={src} />
      <AvatarFallback className={cn(`text-white ${bgColor}`, fallbackClass)}>
        {fallback}
      </AvatarFallback>
    </AvatarRoot>
  );
};

export { AvatarRoot, AvatarImage, AvatarFallback, Avatar };
