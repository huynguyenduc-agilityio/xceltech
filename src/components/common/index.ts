export * from './Input';
export * from './Label';
export * from './Form';
export * from './Button';
export * from './Calendar';
export * from './Popover';
export * from './Avatar';
export * from './Textarea';
export * from './Progress';
export * from './Select';
export * from './Checkbox';
export * from './Carousel';
export * from './Dialog';
export { default as DatePicker } from './DatePicker';
export { default as Toast } from './Toast';
