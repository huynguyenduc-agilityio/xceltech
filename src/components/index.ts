export * from './common';
export * from './Form';
export { default as AdditionalFileUpload } from './AdditionalFileUpload';
export { default as Breadcrumb } from './Breadcrumb';
export { default as DocumentUpload } from './DocumentUpload';
export { default as InfoTile } from './InfoTile';
export { default as LeaveTile } from './LeaveTile';
export { default as Navbar } from './Navbar';
export { default as OverviewCard } from './OverviewCard';
export { default as SideMenu } from './SideMenu';
export { default as Sidebar } from './Sidebar';
export { default as TextField } from './TextField';
export { default as NotificationBadge } from './NotificationBadge';
export { default as Status } from './Status';
export { default as ImageUpload } from './ImageUpload';
export { default as ConfirmDialog } from './ConfirmDialog';
export { default as Table } from './Table';
export { default as SearchBar } from './SearchBar';
export { default as Fallback } from './Fallback';
export { default as FilterMenu } from './FilterMenu';
export { default as ActionsDropdown } from './ActionsDropdown';
export { default as Pagination } from './Pagination';
export { default as LeaveNavbar } from './LeaveNavbar';
export { default as InfoCardWrapper } from './InfoCardWrapper';
export { default as DropdownNotification } from './DropdownNotification';
export { default as DropdownExport } from './DropdownExport';
