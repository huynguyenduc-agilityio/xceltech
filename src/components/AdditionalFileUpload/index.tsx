import { ChangeEvent, useEffect, useRef, useState } from 'react';
import { FieldErrors } from 'react-hook-form';

// Types
import { UploadFileForm } from '@/types';

// Components
import { FormItem, FormMessage, Input, Button } from '@/components';

export interface AdditionalFileUploadProps {
  name: string;
  value?: File | string | null;
  errors?: FieldErrors<UploadFileForm>;
  onFileChange: (file: File | null) => void;
}

const AdditionalFileUpload = ({
  name,
  value,
  errors,
  onFileChange,
}: AdditionalFileUploadProps) => {
  const fileInputRef = useRef<HTMLInputElement | null>(null);
  const [fileUrl, setFileUrl] = useState<string | null>(null);

  const handleFileChange = (e: ChangeEvent<HTMLInputElement>) => {
    const file = e.target.files?.[0] || null;

    onFileChange(file);
  };

  const handleOpenFile = () => {
    fileInputRef.current?.click();
  };

  useEffect(() => {
    if (value instanceof File) {
      const url = URL.createObjectURL(value);

      setFileUrl(url);

      return () => {
        URL.revokeObjectURL(url);
      };
    }

    if (typeof value === 'string') {
      setFileUrl(value);
    }

    if (!value) {
      setFileUrl(null);
    }
  }, [value]);

  const fileName =
    typeof value === 'string'
      ? (value.split('/').pop() ?? '')
      : value instanceof File
        ? value.name
        : '';

  return (
    <FormItem>
      <div className="relative flex items-center h-[68px]">
        <Input
          ref={fileInputRef}
          id={`file-upload-${name}`}
          type="file"
          className="hidden"
          onChange={handleFileChange}
          accept=".pdf"
        />

        <div className="absolute inset-0 flex items-center px-6 w-full h-full bg-blue-light rounded-[15px]">
          <span className="mr-[291px] text-md truncate text-black-smoky-">
            {fileUrl ? (
              <a href={fileUrl} target="_blank" rel="noopener noreferrer">
                {fileName}
              </a>
            ) : (
              ''
            )}
          </span>
        </div>

        <Button
          type="button"
          variant="secondary"
          className="absolute bottom-0 right-0 w-[291px] py-[12px] h-full rounded-[15px]"
          onClick={handleOpenFile}
        >
          Upload
        </Button>
      </div>

      {errors?.file && (
        <FormMessage className="">
          <div className="flex flex-col gap-4 text-red-500 text-sm">
            <p>{errors.file.message}</p>
            <p>Please select again!!!</p>
          </div>
        </FormMessage>
      )}
    </FormItem>
  );
};

export default AdditionalFileUpload;
