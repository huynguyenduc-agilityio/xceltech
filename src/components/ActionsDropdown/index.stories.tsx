import type { Meta, StoryObj } from '@storybook/react';

// Icons
import { CircleDownIcon } from '@/icons';

// Components
import ActionsDropdown from '.';
import { Button } from '../common';

const defaultProps = {
  items: [
    {
      key: 'approve',
      label: 'Approve',
      onClick: () => console.log('Approved'),
    },
    {
      key: 'decline',
      label: 'Decline',
      onClick: () => console.log('Declined'),
      disabled: true,
    },
    {
      key: 'view',
      label: 'View Details',
      onClick: () => console.log('View Details'),
    },
  ],
};

const meta = {
  title: 'Components/ActionsDropdown',
  component: ActionsDropdown,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
  render: (args) => (
    <div className="flex justify-center w-[1000px] p-9 bg-blue-light">
      <ActionsDropdown {...args}>
        <Button className="w-[195px] h-14 shadow-md">
          Export
          <CircleDownIcon className="ml-4" />
        </Button>
      </ActionsDropdown>
    </div>
  ),
} satisfies Meta<typeof ActionsDropdown>;

export default meta;

type Story = StoryObj<typeof ActionsDropdown>;

export const Default: Story = {
  args: { ...defaultProps },
};
