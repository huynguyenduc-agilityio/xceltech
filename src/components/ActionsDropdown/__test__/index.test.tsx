import { render } from '@testing-library/react';

// Icons
import { CircleDownIcon } from '@/icons';

// Components
import { Button } from '@/components/common';
import ActionsDropdown from '..';

describe('ActionsDropdown component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <ActionsDropdown
        items={[
          { label: 'Approve', onClick: () => console.log('Approved') },
          {
            label: 'Decline',
            onClick: () => console.log('Declined'),
            disabled: true,
          },
          {
            label: 'View Details',
            onClick: () => console.log('View Details'),
          },
        ]}
      >
        <Button className="w-[195px] h-14 shadow-md">
          Export
          <CircleDownIcon className="ml-4" />
        </Button>
      </ActionsDropdown>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
