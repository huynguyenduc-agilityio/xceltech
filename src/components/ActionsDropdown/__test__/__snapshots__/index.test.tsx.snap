// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionsDropdown component should match snapshot 1`] = `
<DocumentFragment>
  <button
    aria-expanded="false"
    aria-haspopup="menu"
    class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-primary text-white disabled:hover:bg-primary py-6 w-[195px] h-14 shadow-md"
    data-state="closed"
    id="radix-:r0:"
    type="button"
  >
    Export
    <svg
      class="ml-4"
      fill="none"
      height="14"
      viewBox="0 0 17 14"
      width="17"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#a)"
      >
        <path
          d="M16.734 7c0 3.746-3.685 6.781-8.234 6.781S.266 10.746.266 7 3.95.219 8.5.219 16.734 3.254 16.734 7m-7.67 3.114 4.5-3.705c.312-.257.312-.672 0-.927l-.565-.464a.92.92 0 0 0-1.126 0L8.5 7.796 5.127 5.018a.92.92 0 0 0-1.126 0l-.564.464c-.313.257-.313.673 0 .927l4.499 3.706a.92.92 0 0 0 1.128 0"
          fill="#fff"
        />
      </g>
      <defs>
        <clippath
          id="a"
        >
          <path
            d="M0 0h17v14H0z"
            fill="#fff"
          />
        </clippath>
      </defs>
    </svg>
  </button>
</DocumentFragment>
`;
